File: Frontend/lib/services/request.dart


import 'dart:convert';

import 'package:http/http.dart';

class RequestUnit{

  String patient,date,time,hospital,district,units,bystander,phone,bloodGroup,url="<link>/request.php";

  RequestUnit({this.patient,this.date,this.time,this.hospital,this.district,this.units,this.bystander,this.phone,this.bloodGroup});
  Response response;

  Future<dynamic> registerRequest() async{
    try {
      response = await post(url, body: {
        "patient": patient,
        "date": date,
        "time": time,
        "hospital": hospital,
        "district": district,
        "units": units,
        "bystander": bystander,
        "contact": phone,
        "blood": bloodGroup
      });

      return (response.body);
    }
    catch(error){
      print(error);
      return "error";
    }
  }

}

File: Backend(php)/request.php
<?php
    try{
        require 'connect.php';
        $name=$_POST['patient'];
        $date=$_POST['date'];
        $time=$_POST["time"];
        $hospital=$_POST["hospital"];
        $district=$_POST["district"];
        $units=$_POST["units"];
        $bystander=$_POST["bystander"];
        $contact=$_POST["contact"];
        $bloodGroup=$_POST["blood"];

        $sql="Insert into request(id,pname,bgroup,district,hospital,bname,bphone,date,time,units) Values(NULL,'$name','$bloodGroup','$district','$hospital','$bystander','$contact','$date','$time','$units')";

        if($conn->query($sql) === TRUE){
            echo "Done";
        }
        else{
            echo "error";
        }

        $conn->close();
    }
    catch(Exception $error){
        echo "error";
    }
?>